sig-network/link-state-proposal.md

```diff
@@ -0,0 +1,191 @@
+Adding Link State Management for vNICs
+=
+
+# Overview
+
+The purpose of this design proposal is to introduce support for link state management (up/down) in KubeVirt. This feature
+will enable VM owners to dynamically control the link state. It aligns KubeVirt with traditional virtualization platforms
+and enhances its utility in environments requiring precise network state control.
+
+By introducing link state management, KubeVirt will:
+- Provide VM-level controls to configure the link state of individual interfaces.
+- Report link state info via the VirtualMachineInterface status.
+
+This proposal outlines the necessary changes to KubeVirt, including API changes.
+
+## Motivation
+
+Currently, KubeVirt does not support modifying or reporting the link state of virtual network interfaces.
+This limitation can lead to challenges in implementing advanced networking scenarios, such as:
+- Simulating link failures for testing and debugging.
+- Managing multi-network setups, where some interfaces need to be intentionally disabled.
+
+## Goals
+
+1. Allow users to configure the link state (up/down) of individual interfaces through the KubeVirt API.
+2. Enable link state configuration at both VM creation and runtime.
+3. Report the current link state of each VM interface.
+
+## Non Goals
+
+1. Affect network plumbing on the host - the setup process will remain unchanged change.
+
+## Definition of Users
+
+- VM owner
+
+## User Stories
+
+As a VM owner I would like to be able to:
+
+- Start the VM with one or more interfaces with their link set to `down`.
+- Toggle the link state of one or more interfaces of a running VM.
+- Hot plug an interface with a link state set to `down`.
+
+- I want my interfaces link states to persist following a migration.
+
+## Repos
+
+kubevirt/kubevirt.
+
+# Design
+
+## API Addition
+Currently, the `Interface` struct has the `State` field (currently used for hot-unplug):
+```go
+type Interface struct {
+    ...
+    State InterfaceState `json:"state,omitempty"`
+    ...
+}
+```
+
+Type `InterfaceState` is an enum, currently having a single value:  
+```go
+type InterfaceState string
+
+const (
+	InterfaceStateAbsent InterfaceState = "absent"
+)
+```
+
+The following values will be added to specify the required link state:
+```go
+const (
+    InterfaceStateLinkUp   InterfaceState = "linkUp"
+    InterfaceStateLinkDown InterfaceState = "linkDown"
```

Comment:
I would avoid having "link" in the naming at this abstraction layer, having the naming more generic doesnt make me think whether there are some other options affecting other stuff aspects of the interface (e.g.: network-manager device and link, disabling the link does mean device is disabled).

Same goes for the new attribute in the status, calling it just "state".

Reply:
The existing enumeration is `absent`, which relates to the hot-unplug functionality.
By having the `link` prefix, we relate the new enumerations to the proposed functionality (setting the link state).

Reply:
It can be done without the link in the naming, with simple names.
I dont think it contradicts with "absent" semantics.

Reply:
Changed the possible values to `up` and `down`.